multinode_integration_tests/tests/verify_bill_payment_utils/utils.rs

```diff
@@ -0,0 +1,1003 @@
+                .expect("Mandatory input not provided: consuming node initial service fee balance"),
+        };
+        let mut serving_nodes = [
+            ServingNodeByName::ServingNode1,
+            ServingNodeByName::ServingNode2,
+            ServingNodeByName::ServingNode3,
+        ]
+        .into_iter()
+        .map(|serving_node_by_name| {
+            let debt = debts.remove(0);
+            let ui_port_opt = serving_nodes_ui_ports_opt.remove(0);
+            ServingNodeProfile {
+                serving_node_by_name,
+                debt,
+                ui_port_opt,
+            }
+        })
+        .collect::<Vec<_>>();
+        let node_profiles = NodeProfiles {
+            consuming_node,
+            serving_nodes: core::array::from_fn(|_| serving_nodes.remove(0)),
+        };
+
+        TestInputs {
+            payment_thresholds_all_nodes: self
+                .payment_thresholds_all_nodes_opt
+                .expect("Mandatory input not provided: payment thresholds"),
+            node_profiles,
+        }
+    }
+
+    fn resolve_ports(ui_ports_opt: Option<Ports>) -> (Option<u16>, [Option<u16>; 3]) {
+        match ui_ports_opt {
+            Some(ui_ports) => {
+                let mut ui_ports_as_opt =
+                    ui_ports.serving_nodes.into_iter().map(Some).collect_vec();
+                let serving_nodes_array: [Option<u16>; 3] =
+                    core::array::from_fn(|_| ui_ports_as_opt.remove(0));
+                (Some(ui_ports.consuming_node), serving_nodes_array)
+            }
+            None => Default::default(),
+        }
+    }
+}
+
+struct NodeProfiles {
+    consuming_node: ConsumingNodeProfile,
+    serving_nodes: [ServingNodeProfile; 3],
+}
+
+#[derive(Debug, Clone)]
+pub struct ConsumingNodeProfile {
+    ui_port_opt: Option<u16>,
+    gas_price_opt: Option<u64>,
+    initial_tx_fee_balance_minor_opt: Option<u128>,
+    initial_service_fee_balance_minor: u128,
+}
+
+#[derive(Debug, Clone)]
+pub struct ServingNodeProfile {
+    serving_node_by_name: ServingNodeByName,
+    debt: Debt,
+    ui_port_opt: Option<u16>,
+}
+
+pub struct AssertionsValues {
+    pub final_consuming_node_transaction_fee_balance_minor: u128,
+    pub final_consuming_node_service_fee_balance_minor: u128,
+    pub final_service_fee_balances_by_serving_nodes: FinalServiceFeeBalancesByServingNodes,
+}
+
+pub struct FinalServiceFeeBalancesByServingNodes {
+    balances: [u128; 3],
+}
+
+impl FinalServiceFeeBalancesByServingNodes {
+    pub fn new(node_1: u128, node_2: u128, node_3: u128) -> Self {
+        let balances = [node_1, node_2, node_3];
+        Self { balances }
+    }
+}
+
+pub struct BlockchainParams {
+    chain: Chain,
+    server_url: String,
+    contract_owner_addr: Address,
+    contract_owner_wallet: Wallet,
+    seed: Seed,
+}
+
+struct BlockchainInterfaces {
+    standard_blockchain_interface: Box<dyn BlockchainInterface>,
+    web3: Web3<Http>,
+}
```

Comment:
This structure stores both the interface and the transport, the transport is specific to EVM based blockchains, so perhaps we can use a different name to identify this structure. If it's feasible you can have one object that provides all the blockchain methods together. Something like `ExtendedBlockchainInterface`. You can implement methods on that structure which will help you with all the blockchain interactions that you need for the tests.